src/global_distrib/mod_global_distrib_utils.erl

```diff
@@ -200,8 +200,14 @@ maybe_update_mapping(#jid{luser = <<>>, lserver = LServer} = From, Acc) ->
 maybe_update_mapping(From, Acc) ->
     case mod_global_distrib_mapping:for_jid(From) of
         error ->
-            Origin = mod_global_distrib:get_metadata(Acc, origin),
-            mod_global_distrib_mapping:insert_for_jid(From, Origin);
+            case mod_global_distrib:get_metadata(Acc, origin) of
+                %% Lack of 'global_distrib' indicates 100% local routing...
```

Comment:
How did we get to `maybe_update_mapping` with non-GD message? 😕

Reply:
When `mongoose_router_external*` are moved to the beginning of router list and the message is routed locally in cluster, a packet to component will be processed by `packet_to_component` hook, which is handled by `mod_global_distrib_mapping`.

Reply:
Guess I missed the change to the routing list.

Reply:
The change is done only in one of the deployments, it won't be a default for open source.